sdk/src/settings/builder.rs

```diff
@@ -467,10 +467,38 @@ pub struct BuilderSettings {
     /// See more information on the difference between created vs gathered assertions in the spec here:
     /// <https://spec.c2pa.org/specifications/specifications/2.2/specs/C2PA_Specification.html#_fields>
     pub created_assertion_labels: Option<Vec<String>>,
-
     /// Whether to generate a C2PA archive (instead of zip) when writing the manifest builder.
     /// This will eventually become the default behavior.
     pub generate_c2pa_archive: Option<bool>,
+    /// Whether to auto-generate a [`TimeStamp`] assertion for [`TimeStampFetchScope`] of the manifest if one
+    /// does not already exist.
+    ///
+    /// Useful when a manifest was signed offline and you want to attach a trusted timestamp to it later.
+    ///
+    /// Note that for this setting to take effect, a timestamping authority URL must be set in the
+    /// [`Signer::time_authority_url`]. If the signer is acquired from settings via [`Settings::signer`],
+    /// the URL can be set in [`SignerSettings`].
+    ///
+    /// The default value is None.
+    ///
+    /// [`TimeStamp`]: crate::assertions::TimeStamp
+    /// [`Signer::time_authority_url`]: crate::Signer::time_authority_url
+    /// [`Settings::signer`]: crate::settings::signer
+    /// [`SignerSettings`]: crate::settings::signer::SignerSettings
+    pub timestamp_assertion_fetch_scope: Option<TimeStampFetchScope>,
+}
+
+/// The scope of manifests to fetch timestamps for.
+///
+/// See [`BuilderSettings::timestamp_assertion_fetch_scope`] for more information.
+#[derive(Copy, Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[cfg_attr(feature = "json_schema", derive(JsonSchema))]
+#[serde(rename_all = "lowercase")]
+pub enum TimeStampFetchScope {
+    /// Fetch timestamps for only the parent manifest.
+    Parent,
```

Comment:
There could/should be one more option here.  That is generate a timestamp for any manifest without a timestamp.

Reply:
Maybe I can add a separate option `overwrite_existing_timestamp_assertions: bool`. Or perhaps it would be better like this:
```
auto_timestamp_assertion: {
  enabled: bool,
  skip_existing: bool,
  fetch_scope: Scope,
}
```